src/components/item/PinModal.vue

```diff
@@ -0,0 +1,302 @@
+<template>
+  <b-modal
+    :id="modalId"
+    :title="$t('record.actions.pin')"
+    hide-footer
+    hide-header-close
+    :static="modalStatic"
+    @show="fetchPinningData"
+  >
+    <b-button
+      v-for="(entity, index) in allRelatedEntities"
+      :key="index"
+      :disabled="!fetched"
+      :pressed="selected === entity.id"
+      :data-qa="`pin item to entity choice`"
+      class="btn-collection w-100 text-left d-flex"
+      @click="selectEntity(entity.id)"
+    >
+      <span
+        class="mr-auto"
+      >
+        {{ entity.prefLabel.en }}
+      </span>
+      <span
+        class="icons text-left d-flex justify-content-end"
+      >
+        <span
+          v-if="selected === entity.id"
+          class="icon-check-circle d-inline-flex ml-auto"
+        />
+        <span
+          v-if="pinnedTo(entity.id)"
+          class="icon-push-pin d-inline-flex"
+        />
+      </span>
+    </b-button>
+    <div class="modal-footer">
+      <span class="w-100 help">
+        <span class="icon icon-info-outline d-inline-flex"/>{{ infoText }}
+      </span>
+      <b-button
+        variant="outline-primary"
+        data-qa="cancel button"
+        @click="hide"
+      >
+        {{ $t('entity.actions.cancel') }}
+      </b-button>
+      <b-button
+        v-if="selectedIsPinned || !selectedIsFull"
+        :disabled="!selected"
+        variant="primary"
+        data-qa="toggle pin button"
+        @click="togglePin"
+      >
+        {{ selectedIsPinned ? $t('entity.actions.unpin') : $t('entity.actions.pin') }}
+      </b-button>
+      <span
+        v-else-if="selectedIsFull"
+      >
+        <b-button
+          data-qa="go to set link"
+          :to="$path(selectedLink)"
+          variant="primary"
+        >
+          {{ $t('entity.actions.viewPinned') }}
+        </b-button>
+      </span>
+    </div>
+  </b-modal>
+</template>
+
+<script>
+  import pick from 'lodash/pick';
+  import { mapGetters, mapState } from 'vuex';
+  import makeToastMixin from '@/mixins/makeToast';
+
+  export default {
+    name: 'PinModal',
+
+    mixins: [
+      makeToastMixin
+    ],
+
+    props: {
+      modalId: {
+        type: String,
+        default: 'pin-modal'
+      },
+      entities: {
+        type: Array,
+        required: true
+      },
+      /**
+       * Used for testing, in order to render the modal.
+       */
+      modalStatic: {
+        type: Boolean,
+        default: false
+      }
+    },
+    data() {
+      return {
+        fetched: false,
+        selected: null
+      };
+    },
+
+    computed: {
+      infoText() {
+        if (this.selected) {
```

Comment:
I'm wondering if this conditional could be rewritten. `this.selected` is also been checked in selectedIsPinned and selectedIsFull.